### PR TITLE
feat(query): bump arrow-udf version to enhance types supported by JS UDFs

### PR DESCRIPTION
**Pull request by maxjustus**
I made [some enhancements to the arrow-udf-js code](https://github.com/risingwavelabs/arrow-udf/pull/22) to improve Databend support.
JS UDFs now support Variant, Timestamp, Date and Decimal types in arguments and return values.
This update also enables all intrinsics in rquickjs for better JS compatibility.

I hereby agree to the terms of the CLA available at: https://docs.databend.com/dev/policies/cla/

## Tests

- [ ] Unit Test
- [x] Logic Test
- [ ] Benchmark Test
- [ ] No Test  - _Explain why_

## Type of change

- [ ] Bug Fix (non-breaking change which fixes an issue)
- [x] New Feature (non-breaking change which adds functionality)
- [ ] Breaking Change (fix or feature that could cause existing functionality not to work as expected)
- [ ] Documentation Update
- [ ] Refactoring
- [ ] Performance Improvement
- [ ] Other (please describe):

- - -
This change is [Reviewable](https://reviewable.io/reviews/datafuselabs/databend/15529)

### CODE REVIEW

**Review by sundy-li — APPROVED** [review]

File: Cargo.toml [hunk 1]
```diff
@@ -224,8 +224,8 @@ parquet = { version = "51", features = ["async"] }
 parquet_rs = { package = "parquet", version = "51" }
 
 # Crates from risingwavelabs
-arrow-udf-js = { package = "arrow-udf-js", git = "https://github.com/datafuse-extras/arrow-udf", rev = "a8fdfdd" }
-arrow-udf-wasm = { package = "arrow-udf-wasm", git = "https://github.com/datafuse-extras/arrow-udf", rev = "a8fdfdd" }
+arrow-udf-js = { package = "arrow-udf-js", git = "https://github.com/datafuse-extras/arrow-udf", rev = "d0a21f0" }
```
Comment (sundy-li) [inline Cargo.toml:227]: Maybe we can use the upstream commit now.

**Review by b41sh — APPROVED** [review]
LGTM, thanks for your contribution @maxjustus

**Comment by BohuTANG** [comment]
Unit failed:
```
error: failed to get `arrow-udf-js` as a dependency of package `databend-query v0.1.0 (/workspace/src/query/service)`

Caused by:
  failed to load source for dependency `arrow-udf-js`

Caused by:
  Unable to update https://github.com/datafuse-extras/arrow-udf?rev=d0a21f0#d0a21f0f

Caused by:
  revspec 'd0a21f0fde330a0e5f658a55b58e0405d8372844' not found; class=Reference (4); code=NotFound (-3)
```
https://github.com/datafuselabs/databend/actions/runs/9088408924/job/24977918495?pr=15529

**Comment by maxjustus** [comment]
Ah whoops @BohuTANG - my eyes somehow glossed over the fact that you all use a fork of the main arrow UDF repo. I was using a local crate override while developing this so I didn't notice. It looks like your fork was recently updated to include my changes at that SHA so I suspect that this CI pipeline will pass if you just retry it. I tried to myself just now, but I don't think I have permissions.

**Comment by maxjustus** [comment]
ok @BohuTANG looks like it's all green now!

**Pull request merged**